src/crypto_tools/k256_serde.rs

```diff
@@ -6,16 +6,14 @@
 //! [Implementing Deserialize · Serde](https://serde.rs/impl-deserialize.html)
 
 use ecdsa::elliptic_curve::Field;
-use k256::{
-    elliptic_curve::sec1::{FromEncodedPoint, ToEncodedPoint},
-    EncodedPoint,
-};
+use k256::elliptic_curve::sec1::{FromEncodedPoint, ToEncodedPoint};
 use serde::{de, de::Error, de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
 use zeroize::Zeroize;
 
 use crate::sdk::api::BytesVec;
 
 /// A wrapper for a random scalar value that is zeroized on drop
+/// TODO why not just do this for Scalar below?
```

Comment:
Most scalars are public values, so we don't need to extra work of zeroizing them. I think we should add a `SecretScalar` struct instead that uses the `secrecy` crate as you have mentioned (I would like to not derive `Serialize` at all for those secret values if possible).

Reply:
> (I would like to not derive `Serialize` at all for those secret values if possible).

Unfortunately these threshold protocols like to return a big struct containing secret data.  Users would really like to have `serde` for those structs.

Reply:
Yeah, I'll spend some time thinking about better approaches in the future.